cachix/src/Cachix/Client/Store.hs

```diff
@@ -99,23 +102,23 @@ queryPathInfo store path = do
 
 computeClosure :: Store -> [StorePath] -> IO [StorePath]
 computeClosure store initialPaths = do
-  allPaths <- 
+  allPaths <-
     processGraph (getPath <$> initialPaths) $
       queryPathInfo store >=> \case
         Left _ -> pure []
         Right pathInfo -> pure $ references pathInfo
   return $ StorePath <$> Set.toList allPaths
 
-getStorePathHash :: StorePath -> Text
-getStorePathHash storePath =
-  T.take 32 $ getStorePathBaseName storePath
+getStorePathHash :: Store -> StorePath -> Text
+getStorePathHash store storePath =
+  T.take 32 $ getStorePathBaseName store storePath
 
 getPath :: StorePath -> Text
 getPath (StorePath storePath) = storePath
 
-getStorePathBaseName :: StorePath -> Text
-getStorePathBaseName (StorePath storePath) =
-  T.drop 11 storePath
+getStorePathBaseName :: Store -> StorePath -> Text
+getStorePathBaseName (Store storePrefix _) (StorePath storePath) =
+  T.dropPrefix storePrefix storePath
```

Comment:
The default prefix is `/nix`, so this isn't dropping the `/store/` bit. We might also have to sanitize the prefix to handle any trailing slashes.